docs/secrets.md

```diff
@@ -163,6 +166,38 @@ To reference `secret_foo`inside this file, you can specify it in the inventory a
 
 `secret_foo: ${gpg:components/secrets/mysql_secrets@mysql_passwords.secret_foo}`
 
+### Environment References Backend
+
+It may be useful in some occasions when revealing references to have the values for the reference dynamically
+come from the environment in which Kapitan is executing. This backend provides such functionality. It will attempt
+to locate a value for a reference from the environment using a prefixed variable *`$KAPITAN_VARS_*`* convention
+and use this value with the refs command.
+
+```shell
+$ echo "my_default_value" | kapitan refs --write env:path/to/secret_inside_kapitan -t <target_name> -f -
+```
+
+When this reference is created and then referred to in the parameters, it will use the last path component, from a
+split, to locate a variable in the current environment to use as the value. If this variable cannot be found in the
+environment, it will use the default value written to the refs file on the filesystem.
+
+```yaml
+parameters:
+  mysql_passwordS:
+    secret_foo: ?{env:my/mysql/mysql_secret_foo}
+    secret_bar: ?{env:my/mysql/mysql_secret_bar}
+```
+
+When using the above parameters reference, values would be consulted in the environment from the following
+variables:
+
+* `$KAPITAN_ENV_VARS_mysql_secret_foo`
+* `$KAPITAN_ENV_VARS_mysql_secret_bar`
+
+*Note*: Automatic creation of these references has not been tested. Please ensure the refs file is created before attempting to use it in your parameters structure.
```

Comment:
This point is slightly more crucial and worth testing. Otherwise it might be counterintuitive for users. I can help test this tomorrow too.

Reply:
Okay. I figure it should just work, since I'm just extending off the PlainRef (and I presume that one works for auto-create?) or maybe not.

Reply:
Yup, it should work same as the other backends since autocreation is part of the base ref class. But I can briefly test tomorrow for completeness, no prob. :)

Reply:
I tested both usecases. In detail:
1) No ref created. The ref was `secret_foo: ?{env:my/mysql/mysql_secret_foo||randomstr}`. I ran compile. The secret is auto-generated correctly. Once the secret is revealed if the env var is defined, it displays that as intended.
2) I also tested the use case of the ref already existing and the ref not existing and not having `||randomstr` and both worked as expected. :) One minor issue in the doc.
You mention KAPITAN_ENV_VARS_* but inn code you have `KAPITAN_VAR_*` so let's just make that the same everywhere :) 
The rest LGTM and we can merge after that.

Reply:
So can we remove this from the docs then?